test/test.js

```diff
@@ -347,4 +441,227 @@ describe('rollup-plugin-babel', function() {
+					plugins: ['@babel/transform-runtime'],
+					runtimeHelpers: true,
+				},
+				{ format: 'esm' },
+			).then(({ code }) => {
+				assert.strictEqual(
+					code,
+					`import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
+
+var Foo = function Foo() {
+  _classCallCheck(this, Foo);
+};
+
+export default Foo;
+`,
+				);
+			});
+		});
+
+		it('generates sourcemap by default', () => {
+			return bundleWithOutputPlugin('samples/class/main.js', {}, { sourcemap: true }).then(({ code, map }) => {
+				const target = 'log';
+				const smc = new SourceMapConsumer(map);
+				const loc = getLocation(code, code.indexOf(target));
+				const original = smc.originalPositionFor(loc);
+
+				assert.deepEqual(original, {
+					source: 'samples/class/main.js'.split(path.sep).join('/'),
+					line: 3,
+					column: 10,
+					name: target,
+				});
+			});
+		});
+
+		it('allows using external-helpers plugin even if the externalHelpers flag is not passed', () => {
+			const warnings = [];
+			return bundleWithOutputPlugin(
+				'samples/external-helpers/main.js',
+				{
+					presets: ['@babel/env'],
+					plugins: ['@babel/external-helpers'],
+				},
+				{},
+				{
+					onwarn(warning) {
+						warnings.push(warning.message);
+					},
+				},
+			).then(({ code }) => {
+				assert.deepStrictEqual(warnings, []);
+				assert.ok(code.indexOf('function _classCallCheck') === -1);
+				assert.ok(code.indexOf('babelHelpers.classCallCheck') !== -1);
+				assert.strictEqual(
+					code,
+					`'use strict';
+
+var Foo = function Foo() {
+  babelHelpers.classCallCheck(this, Foo);
+};
+
+var Bar = function Bar() {
+  babelHelpers.classCallCheck(this, Bar);
+};
+
+var main = [new Foo(), new Bar()];
+module.exports = main;
+`,
+				);
+			});
+		});
+
+		it('warns when using the "include" option', () => {
+			const warnings = [];
+			return bundleWithOutputPlugin(
+				'samples/basic/main.js',
+				{
+					include: ['*.js'],
+				},
+				{},
+				{
+					onwarn(warning) {
+						warnings.push(warning.message);
+					},
+				},
+			).then(() => {
+				assert.deepStrictEqual(warnings, [
+					'The "include", "exclude" and "extensions" options are ignored when using the "transformGenerated" option.',
+				]);
+			});
+		});
+
+		it('transforms all chunks in a code-splitting setup', () => {
+			return rollup
+				.rollup({ input: 'samples/chunks/main.js' })
+				.then(bundle => {
+					return bundle.generate({
+						format: 'esm',
+						plugins: [
+							babelPlugin({
+								transformGenerated: true,
+								plugins: ['@babel/syntax-dynamic-import'],
+								presets: ['@babel/env'],
+							}),
+						],
+					});
+				})
+				.then(({ output }) => {
+					assert.deepStrictEqual(
+						output.map(({ code }) => code),
+						[
+							`import('./dep-5f996703.js').then(function (result) {
+  return console.log(result);
+});
+`,
+							`var dep = function dep() {
+  return 42;
+};
+
+export default dep;
+`,
+						],
+					);
+				});
+		});
+	});
```

Comment:
This probably doesnt differ much but we could also add a test for preserveModules case.